// storybook@10.4.3/dist/babel/index.js
import CJS_COMPAT_NODE_URL_3nzk4esy03a from 'node:url';
import CJS_COMPAT_NODE_PATH_3nzk4esy03a from 'node:path';
import CJS_COMPAT_NODE_MODULE_3nzk4esy03a from "node:module";

var __filename = CJS_COMPAT_NODE_URL_3nzk4esy03a.fileURLToPath(import.meta.url);
var __dirname = CJS_COMPAT_NODE_PATH_3nzk4esy03a.dirname(__filename);
var require = CJS_COMPAT_NODE_MODULE_3nzk4esy03a.createRequire(import.meta.url);

// ------------------------------------------------------------
// end of CJS compatibility banner, injected by Storybook's esbuild configuration
// ------------------------------------------------------------
import {
  BabelFileClass,
  babelParse,
  babelParseExpression,
  babelPrint,
  canUpdateVitestConfigFile,
  canUpdateVitestWorkspaceFile,
  core,
  generate,
  getConfigObjectFromMergeArg,
  getEffectiveMergeConfigCall,
  getTargetConfigObject,
  import_core,
  isDefineConfigLike,
  isImportedDefineConfigLikeIdentifier,
  parser,
  parserOptions,
  recast,
  resolveExpression,
  traverse,
  types,
  unwrapTSExpression
} from "../_node-chunks/chunk-YPTE6EJM.js";
import "../_node-chunks/chunk-NU6RNHCQ.js";
import "../_node-chunks/chunk-NP3B2HKL.js";
import "../_node-chunks/chunk-RYS4RAEO.js";
var export_transformSync = import_core.transformSync;
export {
  BabelFileClass,
  babelParse,
  babelParseExpression,
  babelPrint,
  canUpdateVitestConfigFile,
  canUpdateVitestWorkspaceFile,
  core,
  generate,
  getConfigObjectFromMergeArg,
  getEffectiveMergeConfigCall,
  getTargetConfigObject,
  isDefineConfigLike,
  isImportedDefineConfigLikeIdentifier,
  parser,
  parserOptions,
  recast,
  resolveExpression,
  export_transformSync as transformSync,
  traverse,
  types,
  unwrapTSExpression
};
