// storybook/dist/server-errors.js
import CJS_COMPAT_NODE_URL_3nzk4esy03a from 'node:url';
import CJS_COMPAT_NODE_PATH_3nzk4esy03a from 'node:path';
import CJS_COMPAT_NODE_MODULE_3nzk4esy03a from "node:module";

var __filename = CJS_COMPAT_NODE_URL_3nzk4esy03a.fileURLToPath(import.meta.url);
var __dirname = CJS_COMPAT_NODE_PATH_3nzk4esy03a.dirname(__filename);
var require = CJS_COMPAT_NODE_MODULE_3nzk4esy03a.createRequire(import.meta.url);

// ------------------------------------------------------------
// end of CJS compatibility banner, injected by Storybook's esbuild configuration
// ------------------------------------------------------------
import {
  AddonVitestPostinstallConfigUpdateError,
  AddonVitestPostinstallError,
  AddonVitestPostinstallFailedAddonA11yError,
  AddonVitestPostinstallPrerequisiteCheckError,
  AddonVitestPostinstallWorkspaceUpdateError,
  AngularLegacyBuildOptionsError,
  AutomigrateError,
  Category,
  CommonJsConfigNotSupportedError,
  ConflictingStaticDirConfigError,
  CouldNotEvaluateFrameworkError,
  CriticalPresetLoadError,
  FindPackageVersionsError,
  GenerateNewProjectOnInitError,
  GoogleFontsDownloadError,
  GoogleFontsLoadingError,
  IncompatiblePostCssConfigError,
  InvalidFrameworkNameError,
  InvalidStoriesEntryError,
  MainFileEvaluationError,
  MainFileMissingError,
  MinimumReleaseAgeHandledError,
  MissingAngularJsonError,
  MissingBuilderError,
  MissingFrameworkFieldError,
  NoMatchingExportError,
  NoStatsForViteDevError,
  NxProjectDetectedError,
  SavingGlobalSettingsFileError,
  StatusTypeIdMismatchError,
  StorybookError,
  SvelteViteWithSvelteKitError,
  UpgradeStorybookToLowerVersionError,
  UpgradeStorybookUnknownCurrentVersionError,
  ViteModuleGraphSubscriptionError,
  WebpackCompilationError,
  WebpackInvocationError,
  WebpackMissingStatsError
} from "./_node-chunks/chunk-SJTHLB7E.js";
import "./_node-chunks/chunk-C4KI5RNF.js";
import "./_node-chunks/chunk-NP3B2HKL.js";
import "./_node-chunks/chunk-RYS4RAEO.js";
export {
  AddonVitestPostinstallConfigUpdateError,
  AddonVitestPostinstallError,
  AddonVitestPostinstallFailedAddonA11yError,
  AddonVitestPostinstallPrerequisiteCheckError,
  AddonVitestPostinstallWorkspaceUpdateError,
  AngularLegacyBuildOptionsError,
  AutomigrateError,
  Category,
  CommonJsConfigNotSupportedError,
  ConflictingStaticDirConfigError,
  CouldNotEvaluateFrameworkError,
  CriticalPresetLoadError,
  FindPackageVersionsError,
  GenerateNewProjectOnInitError,
  GoogleFontsDownloadError,
  GoogleFontsLoadingError,
  IncompatiblePostCssConfigError,
  InvalidFrameworkNameError,
  InvalidStoriesEntryError,
  MainFileEvaluationError,
  MainFileMissingError,
  MinimumReleaseAgeHandledError,
  MissingAngularJsonError,
  MissingBuilderError,
  MissingFrameworkFieldError,
  NoMatchingExportError,
  NoStatsForViteDevError,
  NxProjectDetectedError,
  SavingGlobalSettingsFileError,
  StatusTypeIdMismatchError,
  StorybookError,
  SvelteViteWithSvelteKitError,
  UpgradeStorybookToLowerVersionError,
  UpgradeStorybookUnknownCurrentVersionError,
  ViteModuleGraphSubscriptionError,
  WebpackCompilationError,
  WebpackInvocationError,
  WebpackMissingStatsError
};
