// storybook/dist/telemetry/index.js
import CJS_COMPAT_NODE_URL_3nzk4esy03a from 'node:url';
import CJS_COMPAT_NODE_PATH_3nzk4esy03a from 'node:path';
import CJS_COMPAT_NODE_MODULE_3nzk4esy03a from "node:module";

var __filename = CJS_COMPAT_NODE_URL_3nzk4esy03a.fileURLToPath(import.meta.url);
var __dirname = CJS_COMPAT_NODE_PATH_3nzk4esy03a.dirname(__filename);
var require = CJS_COMPAT_NODE_MODULE_3nzk4esy03a.createRequire(import.meta.url);

// ------------------------------------------------------------
// end of CJS compatibility banner, injected by Storybook's esbuild configuration
// ------------------------------------------------------------
import {
  ErrorCollector,
  addToGlobalContext,
  cleanPaths,
  computeStorybookMetadata,
  detectAgent,
  getStorybookMetadata,
  isExampleStoryId,
  isTelemetryModuleEnabled,
  isTelemetryStateResolved,
  metaFrameworks,
  onPayloadError,
  oneWayHash,
  removeAnsiEscapeCodes,
  sanitizeAddonName,
  sanitizeError,
  setTelemetryEnabled,
  telemetry
} from "../_node-chunks/chunk-7NHIPC5Z.js";
import {
  SESSION_TIMEOUT,
  getSessionId
} from "../_node-chunks/chunk-7HUVSFHY.js";
import "../_node-chunks/chunk-ADQFWPTE.js";
import "../_node-chunks/chunk-QTJ23TAE.js";
import "../_node-chunks/chunk-DDXC5KYD.js";
import "../_node-chunks/chunk-DUA3THVM.js";
import "../_node-chunks/chunk-6GLEZUPS.js";
import {
  getLastEvents,
  getPrecedingUpgrade,
  isStoryCreatedByAISetup,
  isWithinInitialSession
} from "../_node-chunks/chunk-HM6QFLRG.js";
import "../_node-chunks/chunk-7LMLRE2Q.js";
import "../_node-chunks/chunk-66M6CYI3.js";
import "../_node-chunks/chunk-Y5XIETER.js";
import "../_node-chunks/chunk-SOCOANKH.js";
import "../_node-chunks/chunk-IROO3KK3.js";
import "../_node-chunks/chunk-44YWME5V.js";
import "../_node-chunks/chunk-C4KI5RNF.js";
import "../_node-chunks/chunk-NP3B2HKL.js";
import "../_node-chunks/chunk-VWS27CF2.js";
import "../_node-chunks/chunk-W6G6WTNU.js";
import "../_node-chunks/chunk-RYS4RAEO.js";
export {
  ErrorCollector,
  SESSION_TIMEOUT,
  addToGlobalContext,
  cleanPaths,
  computeStorybookMetadata,
  detectAgent,
  getLastEvents,
  getPrecedingUpgrade,
  getSessionId,
  getStorybookMetadata,
  isExampleStoryId,
  isStoryCreatedByAISetup,
  isTelemetryModuleEnabled,
  isTelemetryStateResolved,
  isWithinInitialSession,
  metaFrameworks,
  onPayloadError,
  oneWayHash,
  removeAnsiEscapeCodes,
  sanitizeAddonName,
  sanitizeError,
  setTelemetryEnabled,
  telemetry
};
