// storybook@10.4.3/dist/cli/index.js
import CJS_COMPAT_NODE_URL_3nzk4esy03a from 'node:url';
import CJS_COMPAT_NODE_PATH_3nzk4esy03a from 'node:path';
import CJS_COMPAT_NODE_MODULE_3nzk4esy03a from "node:module";

var __filename = CJS_COMPAT_NODE_URL_3nzk4esy03a.fileURLToPath(import.meta.url);
var __dirname = CJS_COMPAT_NODE_PATH_3nzk4esy03a.dirname(__filename);
var require = CJS_COMPAT_NODE_MODULE_3nzk4esy03a.createRequire(import.meta.url);

// ------------------------------------------------------------
// end of CJS compatibility banner, injected by Storybook's esbuild configuration
// ------------------------------------------------------------
import {
  ANGULAR_JSON_PATH,
  AddonVitestService,
  AngularJSON,
  ProjectType,
  SUPPORTED_ESLINT_EXTENSIONS,
  addToDevDependenciesIfNotPresent,
  adjustTemplate,
  cliStoriesTargetPath,
  coerceSemver,
  configureEslintPlugin,
  configureFlatConfig,
  copyTemplate,
  copyTemplateFiles,
  extractEslintInfo,
  findEslintFile,
  getBabelDependencies,
  getRendererDir,
  getVersionSafe,
  hasStorybookDependencies,
  normalizeExtends,
  readFileAsJson,
  suggestESLintPlugin,
  writeFileAsJson
} from "../_node-chunks/chunk-W2YN4DWA.js";
import "../_node-chunks/chunk-YPTE6EJM.js";
import {
  detectPnp
} from "../_node-chunks/chunk-XAXODSDY.js";
import "../_node-chunks/chunk-NU6RNHCQ.js";
import "../_node-chunks/chunk-SJTHLB7E.js";
import {
  Settings,
  _clearGlobalSettings,
  globalSettings
} from "../_node-chunks/chunk-66M6CYI3.js";
import "../_node-chunks/chunk-Y5XIETER.js";
import "../_node-chunks/chunk-SOCOANKH.js";
import "../_node-chunks/chunk-IROO3KK3.js";
import "../_node-chunks/chunk-44YWME5V.js";
import "../_node-chunks/chunk-C4KI5RNF.js";
import "../_node-chunks/chunk-NP3B2HKL.js";
import "../_node-chunks/chunk-RYS4RAEO.js";
export {
  ANGULAR_JSON_PATH,
  AddonVitestService,
  AngularJSON,
  ProjectType,
  SUPPORTED_ESLINT_EXTENSIONS,
  Settings,
  _clearGlobalSettings,
  addToDevDependenciesIfNotPresent,
  adjustTemplate,
  cliStoriesTargetPath,
  coerceSemver,
  configureEslintPlugin,
  configureFlatConfig,
  copyTemplate,
  copyTemplateFiles,
  detectPnp,
  extractEslintInfo,
  findEslintFile,
  getBabelDependencies,
  getRendererDir,
  getVersionSafe,
  globalSettings,
  hasStorybookDependencies,
  normalizeExtends,
  readFileAsJson,
  suggestESLintPlugin,
  writeFileAsJson
};
