// storybook@10.4.3/dist/node-logger/index.js
import CJS_COMPAT_NODE_URL_3nzk4esy03a from 'node:url';
import CJS_COMPAT_NODE_PATH_3nzk4esy03a from 'node:path';
import CJS_COMPAT_NODE_MODULE_3nzk4esy03a from "node:module";

var __filename = CJS_COMPAT_NODE_URL_3nzk4esy03a.fileURLToPath(import.meta.url);
var __dirname = CJS_COMPAT_NODE_PATH_3nzk4esy03a.dirname(__filename);
var require = CJS_COMPAT_NODE_MODULE_3nzk4esy03a.createRequire(import.meta.url);

// ------------------------------------------------------------
// end of CJS compatibility banner, injected by Storybook's esbuild configuration
// ------------------------------------------------------------
import {
  CLI_COLORS,
  ConsoleLogger,
  StyledConsoleLogger,
  colors,
  createHyperlink,
  deprecate,
  import_npmlog,
  logTracker,
  logger,
  once,
  prompt,
  protectUrls
} from "../_node-chunks/chunk-6GLEZUPS.js";
import "../_node-chunks/chunk-NP3B2HKL.js";
import "../_node-chunks/chunk-RYS4RAEO.js";
var export_instance = import_npmlog.default;
export {
  CLI_COLORS,
  ConsoleLogger,
  StyledConsoleLogger,
  colors,
  createHyperlink,
  deprecate,
  export_instance as instance,
  logTracker,
  logger,
  once,
  prompt,
  protectUrls
};
